### DISCUSSION TITLE
Actions Workflow : run is limited to 35 days.

### DISCUSSION THREAD

**Discussion by ppmgitrepo** [community/community#48790] [Actions]
### Select Topic Area

Product Feedback

### Body

A Workflow run time is limited to 35 days at max, this is a problem for long running projects. 
In our case a WF spans a CI/CD pipe of SIT>UAT>Production.  Large projects can span > 8 months from Dev to Production. 
Files do not get pushed individually to Production ad hoc, rather the entire project has to clear SIT and UAT and then go to Production. 
By this time the Workflow would have expired due to the above rule. 
If there is a workaround, kindly do let me know.

**Comment by airtower-luna** [comment]
Why do you need the workflow to keep running during development?

If you need a staging/testing environment don't have the staging environment sit there in the runner as part of your workflow, instead the workflow should deploy *to* the staging environment and then end. Obviously the details will vary depending on your software and how you use the staging environment, but that's the general idea.

**Reply by ppmgitrepo** [reply]
Because this is ERP we are creating a CI/CD pipeline on. A project can consist of maybe 600+ files and they are all independent, not patched like a cluster .war or .jar . All of them need to get deployed on SIT , a whole testing cycle takes place, Sign off is got , and they are then deployed to the next UAT env, some files among these may get revved up mid testing. Eventually the whole set needs to move to Production. So a typical configuration is 
Job1 on SIT > Job2 on UAT > Job3 on Prod. 

We may have to change the whole packaging and flow due to this above constraint but for now this is how it is. We have an enabler tool that captures each WF run ID and stores it separate , so that the same Run ID can be picked up and progressed to the next step.  So in effect most of the time gap is waiting between Env's for the approval.

**Comment by ppmgitrepo** [comment]
re-architected the workflows.

**Reply by jsoref** [reply]
It seems like `environments` would be a good replacement for a long running workflow.

I'd be curious to hear what you actually did...
